storm-core/src/clj/backtype/storm/daemon/executor.clj

```diff
@@ -526,7 +526,8 @@
                                                                         out-tuple
                                                                         overflow-buffer)
                                                            ))
-                                         (if rooted?
+                                         (if (and rooted?
+                                                  (seq out-ids)) ;; not empty
```

Comment:
Yes, this is more work than we need. It creates an [IteratorSeq](https://github.com/clojure/clojure/blob/d6baf6e1fb6e2b9ab3e3a459de5ab1d4381938aa/src/jvm/clojure/lang/IteratorSeq.java#L32-L37) each time, since the input is an ArrayList. I'll fix it.